code/obj/item/implant.dm

```diff
@@ -890,6 +901,401 @@ ABSTRACT_TYPE(/obj/item/implant/revenge)
+	var/linked_address = null
+	/// A string that's (usually) unique to each implant. Signals must provide the correct passkey to issue commands to the implant.
+	var/passkey = null
+	/// If TRUE, this implant is burned out and permanently unusable.
+	var/burned_out = FALSE
+	/// The implant's heat level, increased by various actions. Slowly reduces over time.
+	var/heat = 0
+	/// The implant's previous heat level, set after it's adjusted.
+	/// This is used so that the implant can send an alert signal when it enters the danger zone for the first time.
+	var/prev_heat = 0
+	/// The implant's heat dissipation. `heat` is reduced by this value every processing tick.
+	/// The value slowly ramps up over time, but is reset upon being activated. This makes short-term overuse very punishing,
+	/// but allows it to recover decently quickly if given time to rest.
+	var/heat_dissipation = 1
+	/// If `heat` is above this value, each activation has a chance to break the implant permanently.
+	var/const/heat_danger_zone = 100
+	/// This is some messy code, but emotes are like that. Anything in this list will not be triggered by the force-emote function.
+	var/list/emote_blacklist = list(
+		"custom", "customv", "customh", "me", "give", "help", "listbasic", "listtarget", "list", "suicide", "uguu", "juggle", "airquote", "airquotes",
+		"faint", "deathgasp", "collapse", "trip", "monologue", "miranda", "birdwell"
+	)
+
+	New()
+		. = ..()
+		var/datum/reagent/R = pick(concrete_typesof(/datum/reagent/fooddrink/alcoholic))
+		src.passkey = lowertext(replacetext(replacetext(R.name, " ", "_"), "'", ""))
+		if (!src.passkey)
+			src.passkey = "IMP-[rand(111, 999)]"
+		// The `uses_radio` variable only adds a sender component, not a two-way one. So we have to do that manually!
+		if (!src.net_id)
+			src.net_id = generate_net_id(src)
+		MAKE_DEFAULT_RADIO_PACKET_COMPONENT(src.net_id, null, src.pda_alert_frequency)
+		processing_items.Add(src)
+
+	disposing()
+		processing_items.Remove(src)
+		..()
+
+	deactivate()
+		..()
+		processing_items.Remove(src)
+
+	process()
+		src.adjust_heat(-src.heat_dissipation)
+		src.heat_dissipation = min(3, src.heat_dissipation + 0.05)
+
+	can_implant(mob/target, mob/user)
+		if (istype(target, /mob/living/critter/wraith/trickster_puppet))
+			boutput(user, SPAN_ALERT("The implanter shows an error: \"TARGET IS SECRETLY A GHOST\". Huh. You didn't know it even checked for that!"))
+			return
+		return ..()
+
+	implanted(mob/M, mob/I)
+		. = ..()
+		if (!src.burned_out)
+			// this is an anti-frustration feature with the goal of both making it easier to reference chat logs to know who is implanted with what,
+			// as well as to make sure players can still use packets if they forget to scan it onto their remote and didn't write down the network data
+			boutput(I, SPAN_NOTICE("You make a mental note that this implant's network ID is <b>[src.net_id]</b> and its passkey is <b>[src.passkey]</b>."))
+
+	receive_signal(datum/signal/signal)
```

Comment:
It would be flavourful if the item used `netpass_syndicate`. I could see it function like airlocks, where an `access_code` is required on top of the `netpass_syndicate`. A balance dev would have to confirm if giving away the `netpass_syndicate` is okay or not though.